software-engineering-policies/CloudDevelopment/CloudDevelopmentPolicy.md

```diff
@@ -0,0 +1,329 @@
+* Improved collaboration and productivity
+* Cost savings through pay-as-you-go pricing models
+
+It also presents challenges such as:
+
+* Security and compliance concerns
+* Operational complexity during migration and integration
+* Potential vendor lock-in risks
+
+## Acceptable Use for Cloud Technologies
+
+The purpose of this section is to ensure that cloud resources are utilized in compliance with relevant laws, regulations, and organizational standards, while promoting efficiency, reliability, and data security.
+
+## Network Security
+
+### Guidance on Security with the Hub and Spoke Model. See [link](https://github.com/UKHO/how-do-i/blob/3373f41de95525cf795df0d9c1aae7ada2dcea86/docs/subscriptions.md?plain=1)
+
+## Recommendations for Container Hosting / Orchestration (Rich Shawley)
+
+### See Container Policy
+
+### Never Run As Root
+
+## Requirements and Recommendations for
+
+### Auditing
+
+### Testing - limit scope to infra testing ?? hand off to test community ?? - RS
+
+### Virus Scanning
+
+### Code Analysis
+
+### Dependency Checking for Cloud Applications
+
+### Infrastructure as Code - Terraform, etc
+
+Terraform is UKHO's primary tool of choise for IaC and should be used for in all cases. Terraform provided multi-cloud provisioning and so is capable of deploying to any cloud provider.
+
+#### Setting up a new project
+
+The Terraform deployment template found here: https://github.com/UKHO/Terraform-Deployment-Template should be used to ensure consistency when starting up any new project. Contact the DDC team if you have any questions.
+
+#### Networking
+
+The DDC team are responsible for the creating and maintaining the networking spoke Terraform for each project. This is managed in a separate GitHub repository to the project terraform and can be found here: https://github.com/UKHO/Azure-Spokes
+
+#### Modules
+
+The DDC team manages a collection of centralised Terraform modules found here: https://github.com/orgs/UKHO/teams/tfmodules/repositories
+These should be used whenever possible.
+
+#### IaC Scanning (Trivy)
+
+Trivy is UKHO's Chosen IaC Scanning tool. All Infrastructure pipelines should include a Trivy Scan stage before deployment that fails the build if misconfigurations or vulnerabilities are detected.
+A Trivy Stage is included in the previously mentioned Terraform deployment template: https://github.com/UKHO/Terraform-Deployment-Template/blob/5ca2e1c18618dad0df84ef8c08aea87a64c3b94e/azure-pipelines.yml#L37
+
+### Secrets Management
+
+#### Centralized Secrets Management
+
+All secrets, credentials, and sensitive data used by applications must be stored in Azure Key Vault.
+   
+#### Access Control
+
+Access to Key Vault must be restricted using Azure RBAC and Key Vault Access Policies, with access granted based on the principle of least privilege.
+
+#### Network Security
+
+Enable Private Endpoints or Service Endpoints for Key Vault to restrict access to trusted networks.
```

Comment:
Possibly more to add here, in Calypso we are trying to get resources to have ip restrictions, KVs are annoying open to public access - 
![image](https://github.com/user-attachments/assets/6989843c-7db5-46fc-bfd9-8be6a4df9eb4)